webrtc.html

```diff
@@ -2537,6 +2537,91 @@
+            "">replaceTrack()</code></dfn> method is invoked, the user agent MUST
+            run the following steps:</p>
+
+            <ol>
+              <li>
+                <p>Let <var>p</var> be a new promise.</p>
+              </li>
+
+              <li>
+                <p>Let <var>withTrack</var> be the argument to this method.</p>
+              </li>
+
+              <li>
+                <p>If <var>withTrack's</var> <code><a href=
+                  "getusermedia.html#dom-mediastreamtrack-kind">kind</a></code>
+                attribute differs from the </var><code><a href=
+                  "getusermedia.html#dom-mediastreamtrack-kind">kind</a></code>
+                of
+                <code><a href="#dom-rtpsender-track">RTCRtpSender.track</a></code>,
+                then reject
+                <var>p</var> with <code>IncompatibleMediaStreamTrackError</code>,
+                return <var>p</var> and abort these steps.</p>
+              </li>
+
+              <li>
+                <p>Run the following steps in parallel:</p>
+              </li>
+              <ol>
+                <li>
+                  Set the
+                  <code><a href="#dom-rtpsender-track">RTCRtpSender.track</a></code>
+                  attribute to <var>withTrack</var>.
+                <li>
+                  <p>Determine if negotiation is needed to transmit
+                  <var>withTrack</var> in place of the sender's existing track.
+                  Ignore what <code>MediaStream</code> the track resides in and the
+                  track's id in this determination.
+                </li>
+                <ul>
+                  <li>
+                    <p>If no negotiation is needed, then the sender MUST seamlessly
+                    switch to transmitting <var>withTrack</var> in place of what
+                    it is sending, without negotiating, and then resolve
+                    <var>p</var> with <var>undefined</var>. On the remote receiving
+                    end, the track MUST maintain its existing id and stream
+                    associations until the connection ends. To accomplish this,
+                    the remote receiver MUST, on any subsequent renegotiations,
+                    use SSRCs to detect tracks that have been renamed in this
+                    manner, and update any stream associations, rather than
+                    treat them as new tracks. The id of the remote track MUST
+                    NOT change in this case.</p>
+                  </li>
+                  <li>
+                    <p>If negotiation is needed, then fire a <a href=
+                      "#event-negotiation">negotiationneeded</a> event at the
+                    <a>RTCPeerConnection</a> associated with this sender.
+                    In this case, <var>p</var> MUST be resolved with
+                    <var>undefined</var> the next time renegotiation is
+                    successfully completed.</p>
```

Comment:
Should there not be text saying that if renegotiation fails, the promise should be rejected? And then the old track should continue flowing (and be returned from sender.track)?
Would we need something similar to the pending SDPs here, a "pending track" that will (or will not) replace the track when negotiation happens?